qiskit/algorithms/eigensolvers/vqd.py

```diff
@@ -0,0 +1,512 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The Variational Quantum Deflation Algorithm for computing higher energy states.
+
+See https://arxiv.org/abs/1805.08138.
+"""
+
+from __future__ import annotations
+from typing import Callable
+import logging
+from time import time
+from warnings import warn
+import numpy as np
+
+from qiskit.circuit import QuantumCircuit
+from qiskit.circuit.library import RealAmplitudes
+from qiskit.opflow.primitive_ops.pauli_op import PauliOp
+from qiskit.quantum_info.operators.base_operator import BaseOperator
+from qiskit.opflow import PauliSumOp
+from qiskit.primitives import BaseEstimator, EstimatorResult
+from qiskit.algorithms.state_fidelities import BaseStateFidelity
+
+from ..list_or_dict import ListOrDict
+from ..optimizers import Optimizer, SLSQP, Minimizer
+from ..variational_algorithm import VariationalAlgorithm, VariationalResult
+from .eigensolver import Eigensolver, EigensolverResult
+from ..minimum_eigen_solvers.vqe import _validate_bounds, _validate_initial_point
```

Comment:
This does not seem like a good idea anymore - the old VQE will be removed in the future and this will have to be revisited. I get this is a common aspect maybe something should be done in conjunction with the new VQE to have some common module/functions that algos can use for such validation?

Reply:
These functions are retained in the new `VQE`. But perhaps we could add them to `qiskit/algorithms/utils`?

Reply:
It seems reasonable to have them in module since it seems, like before, we are going to share them. In VQD they were a copy until some comment - this re-used internal private functions which to me is not  a good idea.

Reply:
mmm adding them to `utils` sounds like a good idea, but if I'm not mistaken, the `VQE` PR should be merged after mine to fix [this](https://github.com/Qiskit/qiskit-terra/pull/8702/files#diff-4c3e32808a9c93b501119564c7f52c29f121f32ddc08c2f51c9a947c493a3c1cR24-R25). What do you think, should I migrate them to `qiskit/algorithms/utils`in this PR to unblock this situation? @declanmillar @woodsp-ibm

Reply:
I think that needs updating but as it stands, although internally its using the existing version it still works - the tests there also are only using PauliSumOp (which is an OperatorBase and the only type supported before) not any BaseOperator types like I mentioned here. The function there is pretty much dependent on this here right. What we could do is merge the VQE one with that still standing and in your PR here fixup the import to put it to the new algorithm here where we want it).